std/algorithm/comparison.d

```diff
@@ -1911,3 +1915,101 @@ bool isPermutation(alias pred = "a == b", Range1, Range2)
         [mytuple(2, 3), mytuple(1, 2)]
     ));
 }
+
+enum ifTestable(T, alias pred = a => a) = __traits(compiles, { if ((pred(T()))) {} }) || isArray!T;
+
+/** Returns: First Parameter $(D p) (element of $(D a)) that passes an $(D if
```

Comment:
Yah, that won't look well. Open with a general description and keep the "Returns:" section small.